Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -423,19 +423,28 @@ bool TryConvert(BindingExpressionPart part, ref object value, Type convertTo, bo
 				return true;
 
 			object original = value;
+			var locale = CultureInfo.InvariantCulture;
 			try
 			{
 				var stringValue = value as string ?? string.Empty;
-				// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
-				// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
-				if (stringValue.EndsWith(".") && DecimalTypes.Contains(convertTo))
-					throw new FormatException();
 
-				// do not canonicalize "-0"; user will likely enter a period after "-0"
-				if (stringValue == "-0" && DecimalTypes.Contains(convertTo))
-					throw new FormatException();
+				if (DecimalTypes.Contains(convertTo))
+				{
+					// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
+					// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
+					if (stringValue.EndsWith("."))
```

Comment:
This logic should probably be updated to test if the string value ends with the `CurrentCulture` decimal separator instead of the hard coded dot. Even if we stick with invariant culture we should replace the hardcoded dot with the `NumberDecimalSeparator`.